radix-engine/src/system/system_callback.rs

```diff
@@ -1571,7 +1571,61 @@ impl<V: SystemCallbackObject> KernelTransactionExecutor for System<V> {
                         )
                     }
                 }
-            };
+            }
+            .and_then(|_| {
+                match hash_nullification {
+                    IntentHashNullification::TransactionIntent { .. } => {
+                        // Transaction intent nullification is historically not costed.
```

Comment:
There is no base fee, but `tx_payload_cost` fee which is proportional to payload size.

```
// The size of a typical transfer transaction is 400 bytes, and the cost will be 400 * 40 = 16,000 cost units
// The max size of a transaction is 1 MiB, and the cost will be 1,048,576 * 40 = 41,943,040 cost units
// This is roughly 1/24 of storing data in substate store per current setup.
mul(cast(size), 40)
```

Reply:
Oh! I thought there was a base fee. Interesting. Perhaps we should add one at Cuttlefish to account for the ident? Or just bill for TransactionIntent nullification from Cuttlefish?

Reply:
We have various uncosted substate reads (such as systemboot, vmboot, kernelboot).

Per the comment above, the minimum transaction cost unit is about `16,000`, which is just enough to cover one  `ReadFromDBNotFound` (which is much more expensive than `ReadFromDB`).

I think we can live with this for now, OR introduce a larger base.

Reply:
What do you think about the system logic tweak to also bill for transaction intent reads? Would be quite straightforward I imagine?

Reply:
> Perhaps we should add one at Cuttlefish to account for the ident

At execution time, we can't tell if it's v1 or v2. 

If we start charging for transaction intent from cuttlefish, fees for all transaction will be increased, which has a user impact. Not too sure about if it's worth to do it now.

Reply:
We're in system here, right? 

o have access to the version number either from version state (current) or from a generic (if we merge https://github.com/radixdlt/radixdlt-scrypto/pull/1919 ).

> If we start charging for transaction intent from cuttlefish, fees for all transaction will be increased, which has a user impact. Not too sure about if it's worth to do it now.

It will, but not by much I imagine?

Reply:
> It will, but not by much I imagine?

It's about 16,000.